knowledge_engine/tests/storage/test_elasticsearch_backend.py

```
@@ -70,6 +72,233 @@ def test_resolve_hybrid_alpha_uses_keyword_weight_when_only_keyword_is_present(
+
+        mock_client = MagicMock()
+        mock_client_class.return_value = mock_client
+
+        backend = ElasticsearchBackend(
+            {
+                "url": "http://localhost:9200",
+                "indexStrategy": {"mode": "per_dataset", "prefix": "test"},
+            }
+        )
+
+        with pytest.raises(ValueError, match="RetrievalScope.document_ids"):
```

Ruff (0.15.20)

[warning] Pattern passed to `match=` contains metacharacters but is neither escaped nor raw

(RUF043)